### PR TITLE
Merge RamSoundFont, FileRenderer, gitignore and other goodies

### PR DESCRIPTION
**Pull request by ChristianRomberg**
Hi @pedromsilvapt.
Seeing that you made some extensive and pretty nice looking additions to pyfluidsynth, I was wondering if you could merge them back into the base repository.
Greetings,
Chris

### CODE REVIEW

**Comment by pedromsilvapt** [comment]
Hi @ChristianRomberg,
Sorry for the late reply. I'd be happy to merge them back, but to be honest I feel the code I did is a little bit of a mess right now, and some small parts are maybe too specific for the usecase I had. I was thinking of maybe creating a specific branch, where we could discuss what too include and what not, as well as allowing me to cleanup some stuff before merging.
What do you think?

**Comment by ChristianRomberg** [comment]
Jeez I am so sorry. I did not mean to ghost this conversation. Of course, that sounds great.

**Comment by pedromsilvapt** [comment]
Don't worry, it's ok. I haven't had much time myself to work on it, but will try to do it soon.

File: pyfluidsynth.py [hunk 1]
```diff
@@ -36,9 +39,9 @@
 
 lib = find_library('fluidsynth') or \
     find_library('libfluidsynth') or \
-    find_library('libfluidsynth-3') or \
```
Comment (cclauss) [inline pyfluidsynth.py:39]: Please revert this change and rebase.  It is best to install the more current library `libfluidsynth-3` before trying earlier libraries (`libfluidsynth-2` and `libfluidsynth-1`).

File: pyfluidsynth.py [hunk 2]
```diff
@@ -107,16 +113,106 @@ def cfunc(name, result, *args):
...
+                                ( 'sample', c_void_p, 1 ),
+                                ( 'name', c_char_p, 1 ) )
+
+    fluid_sample_set_sound_data = cfunc( 'fluid_sample_set_sound_data', c_int,
+                                    ( 'sample', c_void_p, 1 ),
+                                    ( 'data', c_void_p, 1 ),
+                                    ( 'nbframes', c_uint, 1 ),
+                                    ( 'copy_data', c_short, 1 ),
+                                    ( 'rootkey', c_int, 1 ) )
+except:
```
Comment (cclauss) [inline pyfluidsynth.py:194]: Please avoid bare `except`:
* https://peps.python.org/pep-0008/#programming-recommendations
* https://docs.astral.sh/ruff/rules/bare-except/
* https://realpython.com/the-most-diabolical-python-antipattern/

File: setup.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 from setuptools import setup
 
-setup (name = 'pyFluidSynth',
-       version = '1.3.2',
+setup (name = 'pyFluidSynth-musikla',
```
Comment (cclauss) [inline setup.py:3]: # ???

File: setup.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+       version = '1.3.3',
        author = 'Nathan Whitehead',
        author_email = 'nwhitehe@gmail.com',
-       url = 'https://github.com/nwhitehead/pyfluidsynth',
+       url = 'https://github.com/pedromsilvapt/pyfluidsynth',
```
Comment (cclauss) [inline setup.py:7]: # ???

File: pyfluidsynth.py [hunk 5]
```diff
@@ -642,20 +888,24 @@ def __init__(self, gain=0.2, samplerate=44100, channels=256, **kwargs):
         self.setting('synth.gain', gain)
         self.setting('synth.sample-rate', float(samplerate))
         self.setting('synth.midi-channels', channels)
-        for opt,val in kwargs.items():
+        for opt,val in iteritems(kwargs):
```
Comment (cclauss) [inline pyfluidsynth.py:891]: Why go backwards to Python 2?!?

**Review by cclauss — CHANGES_REQUESTED** [review]
Please rebase.

File: pyfluidsynth.py [hunk 6]
```diff
@@ -624,8 +751,127 @@ def fluid_synth_write_s16_stereo(synth, len):
...
+
+        fluid_sample_set_name( sample, create_string_buffer( name.encode(), 20 ) )
+
+        assert wave.getsampwidth() == 2, f"Samples must be 16bit WAVE files, got {wave.getsampwidth() * 8}bit"
+        assert wave.getnchannels() == 1, f"Samples must be mono WAVE files, got {wave.getnchannels()} channels"
+        assert wave.getframerate() == 44100, f"Samples must be 44100hz WAVE files, got {wave.getframerate()}hz"
+
+        nbframes = wave.getnframes()
+
+        if nbframes == 2147483647:
```
Comment (cclauss) [inline pyfluidsynth.py:829]: Please add a comment about where this number comes from.
```suggestion
        if nbframes == 2_147_483_647:
```

**Pull request closed without merging**